tests/cloud_test_logic/cloud_instance_mappings.py

```diff
@@ -0,0 +1,96 @@
+from functools import wraps
+from typing import NamedTuple, Union, List
+from unittest import mock
+
+import requests
+
+
+class GetIndexesIndexResponseObject(NamedTuple):
+    """Represents a response object containing information about an index object in get indexes response.
+
+    This class is designed to encapsulate data related to an index mapping, making it easier
+    to pass this information as an argument to other functions.
+    This class is used in the mock_instance_mappings decorator.
+
+    Example Usage:
+    This will create basic index object with name "test-index", status "READY", and endpoint "endpoint"
+    >>> @mock_get_indexes_response([GetIndexesIndexResponseObject.get_default_index_object()])
+    Response object would look:
+    {"results": [{"index_name": "test-index", "index_status": "READY", "endpoint": "endpoint"}]}
+
+    This will create index object with specified parameters
+    >>> @mock_get_indexes_response([GetIndexesIndexResponseObject("other-index", "CREATING", "endpoint2")])
+    Response object would look:
+    {"results": [{"index_name": "other-index", "index_status": "CREATING", "endpoint": "endpoint2"}]}
+
+"""
+    index_name: str
+    index_status: str
+    endpoint: str
+
+    @staticmethod
+    def get_default_index_object():
+        return GetIndexesIndexResponseObject(
+            index_name="test-index",
+            index_status="READY",
+            endpoint="endpoint"
+        )
+
+
+def mock_get_indexes_response(indexes_list: Union[List[GetIndexesIndexResponseObject], None], to_return_mock: bool = False):
+    """Function decorator to mock the get indexes endpoint.
+
+    This decorator is used to mock the behavior of the requests.get which is used by
+    MarqoCloudInstanceMappings object to retrieve and store specific information about
+    cloud indexes: index_name, index status and index endpoint.
+    It allows you to set up mock responses for requests to the "/indexes" URL
+    while allowing real requests to other URLs.
+
+    Args:
+        indexes_list (Union[List[GetIndexesIndexResponseObject], None]): A list of
+            InstanceMappingIndexData objects to be used for mocking responses
+            to requests to the "/indexes" URL. If None, no mock data will be used.
```

Comment:
Do you mean `if None` then `get_indexes` will return a `get_indexes` response that contains no index information?

Remember to think from the intended audience's perspective